File: apps/api/plane/app/serializers/issue_milestone.py
from rest_framework import serializers
from .base import BaseSerializer
from plane.db.models import IssueMilestoneItem


class IssueMilestoneItemSerializer(BaseSerializer):
    class Meta:
        model = IssueMilestoneItem
        fields = "__all__"
        read_only_fields = [
            "workspace",
            "project",
            "issue",
            "created_by",
            "updated_by",
            "created_at",
            "updated_at",
            "deleted_at",
        ]

File: packages/types/src/milestone/issue-milestone.ts
export interface IIssueMilestoneItem {
  id: string;
  issue: string;
  project: string;
  workspace: string;
  content: string;
  target_date: string | null;
  sort_order: number;
  created_at: string;
  updated_at: string;
  created_by: string;
  updated_by: string;
  deleted_at: string | null;
}

export type TPartialIssueMilestoneItem = Partial<Omit<IIssueMilestoneItem, "issue">>;
